middleware.js

```diff
@@ -61,21 +61,37 @@ async function recordVisitFromEmail (req, res, next) {
     next();
     return;
   }
-  if (!req.query.subscriber_id || !Number.isInteger(Number(req.query.subscriber_id))) {
-    next();
-    return;
+
+  const breachDetailsRE = /breach-details\/(\w*)$/;
+  const capturedMatch = req.path.match(breachDetailsRE);
+
+  // Send engagement ping to FXA if req.query contains a valid subscriber ID
+  if (req.query.subscriber_id && Number.isInteger(Number(req.query.subscriber_id))) {
+    const subscriber = await DB.getSubscriberById(req.query.subscriber_id);
+
+    if (!subscriber.fxa_uid || subscriber.fxa_uid === "") {
+      next();
+      return;
+    }
+    const utmContent = (capturedMatch) ? `&utm_content=${capturedMatch[1]}` : "";
+    const fxaMetricsFlowPath = `metrics-flow?utm_source=${req.query.utm_source}&utm_medium=${req.query.utm_medium}${utmContent}&event_type=engage&uid=${subscriber.fxa_uid}&service=${AppConstants.OAUTH_CLIENT_ID}`;
+    const fxaResult = await FXA.sendMetricsFlowPing(fxaMetricsFlowPath);
+    log.info(`fxaResult: ${fxaResult}`);
   }
-  const subscriber = await DB.getSubscriberById(req.query.subscriber_id);
-  if (!subscriber.fxa_uid || subscriber.fxa_uid === "") {
+
+  // Redirect users who have clicked "Resolve this breach" from an email
+  // and aren't signed in to the /oauth flow.
+  if (req.query.utm_campaign && req.query.utm_campaign === "resolve-this-breach-link") {
```

Comment:
This is true even if the user is signed-in. So, when I was signed in and clicked "Resolve this breach" in my email, I was still redirected thru FxA sign-in. Needs a check with `_getRequestSessionUser` above here.

```js
if (__getRequestSessionUser(req) !== null) {
  next();
  return;
}
```

Reply:
Tried this and kept wheeling out trying to get to "/" when not signed in so I tried something slightly different. You now should be able to, when signed in, click "Resolve this breach" or "Go to dashboard" and not get rerouted to the `/oauth` flow.